Tribler/community/multichain/database.py

```diff
@@ -0,0 +1,314 @@
+""" This file contains everything related to persistence for MultiChain.
+"""
+from os import path
+# Hasher used to generate hashes in the MultiChain Community
+from hashlib import sha1
+
+from Tribler.dispersy.database import Database
+from Tribler.community.multichain.conversion import encode_block
+
+DATABASE_DIRECTORY = path.join(u"sqlite")
+""" Path to the database location + dispersy._workingdirectory"""
+DATABASE_PATH = path.join(DATABASE_DIRECTORY, u"multichain.db")
+""" Version to keep track if the db schema needs to be updated."""
+LATEST_DB_VERSION = 1
+""" Schema for the MultiChain DB."""
+schema = u"""
+CREATE TABLE IF NOT EXISTS multi_chain(
+ block_hash			        TEXT PRIMARY KEY,
+ up                         INTEGER NOT NULL,
+ down                       INTEGER NOT NULL,
+ total_up_requester         INTEGER NOT NULL,
+ total_down_requester       INTEGER NOT NULL,
+ sequence_number_requester  INTEGER NOT NULL,
+ previous_hash_requester	TEXT NOT NULL,
+ total_up_responder         INTEGER NOT NULL,
+ total_down_responder       INTEGER NOT NULL,
+ sequence_number_responder  INTEGER NOT NULL,
+ previous_hash_responder	TEXT NOT NULL,
+ public_key_requester		TEXT NOT NULL,
```

Comment:
I would still like you to consider using mids. Mids are used everywhere in Dispersy, so why differ from this behaviour here. Moreover, they're smaller than public-keys resulting in a slightly smaller database.